File: c#/CommonWeb/WebApplication1/Intf/IUserService.cs
using CommonHelper.Helper.CommonEntity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace CommonWeb.Intf
{
    /// <summary>
    /// 用户操作的业务逻辑接口类，
    /// 所有用户操作的业务逻辑必须实现这些方法。
    /// </summary>
    public interface IUserService
    {
        /// <summary>
        /// 根据签名的signKey获取签名的signSecret
        /// </summary>
        /// <param name="signKey"></param>
        /// <returns></returns>
        string FindSecretByKey(string signKey);

        /// <summary>
        /// 根据signKey加载用户名
        /// </summary>
        /// <param name="signKey"></param>
        /// <returns></returns>
        string LoadUsernameBySignKey(string signKey);

        /// <summary>
        /// 根据用户名，加载用户的树节点
        /// </summary>
        /// <param name="username">用户名</param>
        /// <returns></returns>
        List<LeftMenu> LoadLeftMenus(string username);

        /// <summary>
        /// 登陆用户名、密码校验
        /// </summary>
        /// <param name="username">用户名</param>
        /// <param name="password">密码</param>
        /// <returns></returns>
        bool CheckLogin(string username,string password);
    }
}

File: java/commonWeb/src/main/java/web/template/intf/IUserService.java
package web.template.intf;
import java.util.List;
import com.txj.common.entity.LeftMenu;
/**
 * 用户操作的业务逻辑接口类，
 * 所有用户操作的业务逻辑必须实现这些方法。
 * @author admin
 *
 */
public interface IUserService {
	/**
	 * 根据签名的signKey获取签名的signSecret
	 * @param signKey 签名的key
	 * @return 返回用户的签名密钥
	 */
    String findSecretByKey(String signKey);

    /**
     * 根据signKey加载用户名
     * @param signKey 签名的key
     * @return 返回登陆的用户名
     */
    String loadUsernameBySignKey(String signKey);
    
    /**
     * 根据用户名加载密码
     * @param username 用户名
     * @return 返回登陆密码
     */
    String loadPasswordByUsername(String username);

    /**
     * 根据用户名，加载用户的树节点
     * @param username 用户名
     * @return 返回用户有权操作的树菜单节点
     */
    List<LeftMenu> loadLeftMenus(String username);
}